docs/code_editor/custom_environment_variables.mdx

```diff
@@ -0,0 +1,15 @@
+# Custom Environment Variables
+
+In a self-hosted environment, Windmill allows you to set custom environment variables for your scripts. This feature is useful when a script needs an environment variable prior to the main function executing itself. For instance, some libraries in Go do some setup in the 'init' function that depends on environment variables.
+
+To add a custom environment variable to a script in Windmill, you should follow this format: `<KEY>=<VALUE>`. Where `<KEY>` is the name of the environment variable and `<VALUE>` is the corresponding value of the environment variable.
+
+## Request headers
+
+It is possible for jobs to take request headers as arguments. To do so, either specify in the query args the headers to process at `include_headers`, separated with `,`. e.g: `/api/w/admins/jobs/run_wait_result/p/u/user/undisputed_script?include_header=X-Sign,foo`
+
+or use the env variable: `INCLUDE_HEADERS` with the same format so that all requests to any job will include the headers.
+
+## Raw payload
+
+Similarly to request headers, if the query args contain `raw=true`, then an additional argument will be added: `raw_string` which contains the entire json payload as a string (without any parsing). This is useful to verify the signature of the payload for example (discord require the endpoints to verify the signature for instance).
```

Comment:
This should probably be mentionned in core concepts, close where we define the way we parse arguments